pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -278,21 +292,23 @@ func createIngress(isvc *v1beta1.InferenceService, config *v1beta1.IngressConfig
 		}
 		httpRoutes = append(httpRoutes, &explainerRouter)
 	}
-	// Add predict route
-	httpRoutes = append(httpRoutes, &istiov1alpha3.HTTPRoute{
-		Match: createHTTPMatchRequest("", serviceHost,
-			network.GetServiceHostname(isvc.Name, isvc.Namespace), isInternal, config),
-		Route: []*istiov1alpha3.HTTPRouteDestination{
-			createHTTPRouteDestination(backend, isvc.Namespace, config.LocalGatewayServiceName),
-		},
-		Headers: &istiov1alpha3.Headers{
-			Request: &istiov1alpha3.Headers_HeaderOperations{
-				Set: map[string]string{
-					"Host": network.GetServiceHostname(backend, isvc.Namespace),
+	// Add predict route, only if deployment mode is not ModelMesh
```

Comment:
This is the internal route which is still needed to route to explainer I think, it is not modelmesh specific as the request always hits explainer first which is orchestrated by kserve

Reply:
The condition is to filter out the predict route when predictor is managed by ModelMesh. Since currently ModelMesh doesn't work with ingress, we won't be able to orchestrate explainers to work with ModelMesh predictors automatically in KServe ingress. We might want to think about whether a better solution exists.

Reply:
The default for modelmesh service is grpc. It seems a bit complicated to make it a generic route in ingress to modelmesh, involving getting modelmesh config for service name, port number, payload conversion etc. I would suggest a separate discuss and PR to tackle it.